global/scripts/debug.shlib

```diff
@@ -0,0 +1,328 @@
+#!/bin/echo This script should not run directly:
+
+# Description:
+#
+#   Function for printing debugging information on caught errors.
+#   It makes use of error trapping. 
+#
+#   Default behavior is that error trapping is enabled, 
+#   and the script exits upon trapped error with the same status as the failed command.
+#   
+#   To disable error trapping altogether run `debug_disable_trap`
+#      (Note that if `set -e` is set in the calling script, the script would still exit upon error 
+#       through that mechanism).
+#   To disable exit upon trapped error run `debug_disable_exit`
+#       In this case (trap doesn't exit), execution proceeds to the next line of the script, 
+#       with `$?` still set to whatever value the failed command errored with.
+#
+# Input:
+#
+#   $1 - line number
+#   $2 - command ran
+#   $3 - exit status
+
+debug_on_error()
+{    
+    if [ ! $HCP_DEBUG_USETRAP == 'OFF' ]; then        
+
+        local me=`basename "$0"`
+        local startreport=""
+        local endreport=""
+        local command=$(eval debug_first "$2")
+
+        # --- output a regular HCP error log
+
+        log_Err "'$command' command failed with return code: $3"
+
+        # --- are we printing in color
+
+        if [ "$HCP_DEBUG_COLOR" == "TRUE" ] ; then
+           startreport="\033[33m"
+           endreport="\033[0m"
+        fi
+        
+        # --- output a detailed report if usetrap is not turned off    
+        echo                                                               
+        echo -e "${startreport}===> ERROR: Command returned with nonzero exit code"
+        echo "---------------------------------------------------"
+        echo "         script: $me"                                            
+        echo "stopped at line: $1"                                    
+        echo "           call: $2"
+        echo "  expanded call: $(eval echo "$2")"                            
+        echo "      exit code: $3"                                        
+        echo -e "---------------------------------------------------${endreport}" 
+        echo                                                               
+        
+        # --- Check if interactive shell
```

Comment:
The comment `# --- Check if interactive shell` should be associated with the `if [[ "$-" == *i* ]]` line.  Not above the `if [ ! $HCP_DEBUG_EXIT_ON_ERROR == 'OFF' ]; then` line, which implies that the entire `if [ ! $HCP_DEBUG_EXIT_ON_ERROR == 'OFF' ]; then` is about an interactive shell.

Reply:
Changed.